finagle-core/src/main/scala/com/twitter/finagle/ssl/KeyCredentials.scala

```diff
@@ -48,4 +50,10 @@ object KeyCredentials {
    */
   case class CertKeyAndChain(certificateFile: File, keyFile: File, caCertificateFile: File)
       extends KeyCredentials
+
+  /**
+   *
+   * @param keyManagerFactory
+   */
+  case class FromKeyManager(keyManagerFactory: KeyManagerFactory) extends KeyCredentials
```

Comment:
same sentiment here on the name

Reply:
Calling this class `KeyManagerFactory` as well makes it clash with the argument type. I'm ambivalent in overriding names like that especially when making it usable from Java which does not have import renaming.